src/main/java/org/commcare/cases/entity/NodeEntityFactory.java

```diff
@@ -73,10 +73,7 @@ public Detail getDetail() {
                  * We are wrapping the original exception in a new XPathException to avoid
                  * refactoring large number of functions caused by throwing XPathSyntaxException here.
                  */
-                XPathException xe = new XPathException(e.getMessage(), e);
-                if (e instanceof XPathException) {
-                    xe.setSource(((XPathException)e).getSource());
-                }
+                XPathException xe = new XPathException(e);

```

Comment:
You are right that we only actually want to wrap in case of an XPathSyntaxException though for the sake of consistency I have wrapped both of them. Since we are now not passing message explicitly the new XPathExcpetion will in both cases give the same message as the original exception (Earlier it was duplicating for XPathException). If it's confusing we can throw the XPathException as it is and only wrap in case of a XPathSyntaxException.